src/v/kafka/server/handlers/describe_producers.cc

```diff
@@ -0,0 +1,162 @@
+      [ntp = std::move(ntp)](cluster::partition_manager& pm)
+        -> ss::future<
+          checked<cluster::rm_stm::transaction_set, kafka::error_code>> {
+          auto partition = pm.get(ntp);
+          if (!partition) {
+              co_return kafka::error_code::unknown_topic_or_partition;
+          }
+
+          if (!partition->is_leader()) {
+              co_return kafka::error_code::not_leader_for_partition;
+          }
+
+          auto rm_stm_ptr = partition->rm_stm();
+
+          if (!rm_stm_ptr) {
+              vlog(
+                klog.error,
+                "Can not get rm_stm for ntp({}). Looks like transaction are "
+                "disabled",
+                ntp);
+
+              co_return kafka::error_code::unknown_server_error;
+          }
+
+          auto res = co_await rm_stm_ptr->get_transactions();
+          if (!res.has_value()) {
+              vlog(
+                klog.error,
+                "get_transactions was timeouted. It means node is not leader "
+                "for ntp({}} anymore",
+                ntp);
+              co_return kafka::error_code::not_leader_for_partition;
+          }
+
+          co_return res.value();
+      });
+}
+
+} // namespace
+
+template<>
+ss::future<response_ptr>
+describe_producers_handler::handle(request_context ctx, ss::smp_service_group) {
+    describe_producers_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+
+    describe_producers_response response;
+
+    auto unauthorized_it = std::partition(
+      request.data.topics.begin(),
+      request.data.topics.end(),
+      [&ctx](const topic_request& topic) {
+          return ctx.authorized(security::acl_operation::read, topic.name);
+      });
+
+    std::vector<topic_request> unauthorized(
+      std::make_move_iterator(unauthorized_it),
+      std::make_move_iterator(request.data.topics.end()));
+
+    request.data.topics.erase(unauthorized_it, request.data.topics.end());
+
+    auto kafka_namespace = "kafka";
+
+    for (const auto& topic : request.data.topics) {
+        topic_response topic_resp;
+        topic_resp.name = topic.name;
+        for (const auto& partition : topic.partition_indexes) {
+            partition_response partition_resp;
+            partition_resp.partition_index = partition;
+            model::ntp ntp(kafka_namespace, topic.name, partition);
+            auto txs = co_await get_producers_for_partition(ctx, ntp);
+            if (txs.has_value()) {
+                for (const auto& tx : txs.value()) {
+                    producer_state producer_info;
+                    producer_info.producer_epoch = tx.first.epoch;
+                    producer_info.producer_id = tx.first.id;
+
+                    if (tx.second.info) {
+                        producer_info.last_timestamp
+                          = tx.second.info->last_update.time_since_epoch()
+                              .count();
```

Comment:
Request should contain `epoch` timestamp, the one returned from last_update is not what is expected as rm_stm is using `lowres_clock`

Reply:
Yes. It is problem. Will try with epoch timestamp